tests/benches/promote.c

```diff
@@ -0,0 +1,44 @@
+#include <assert.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+#ifdef DO_PROMOTE
```

Comment:
Why the `#ifdef`?

Reply:
It's a program that's benchmarked with and without promotion, to see what difference it makes (if any!).